### PR TITLE
feat: export transaction history and show xpub (NMA-1257)

### PR DESCRIPTION
**Pull request by HashEngineering**
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/ui/ExportTransactionHistoryDialogBuilder.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import de.schildbach.wallet_test.R;
+import kotlin.Unit;
+
+/**
+ * @author Eric Britten
+ */
+public abstract class ExportTransactionHistoryDialogBuilder extends BaseAlertDialogBuilder {
+    private final Activity context;
+
+    private final EditText viewDescription;
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/ExportTransactionHistoryDialogBuilder.java:62]: This is copied from `CrashReporterDialogBuilder`.  Some things are removed that are not needed.

Future designs will eliminate the need for this dialog, but the code that is related to sharing an attachment with the transaction history will still be needed. 

I suggest that we can update, refactor or rewrite the code related to exporting files in the story will be adding the UI and associated features.

File: wallet/src/de/schildbach/wallet/ui/ToolsActivity.kt [hunk 2]
```diff
@@ -48,5 +70,86 @@ class ToolsActivity : BaseMenuActivity() {
...
+        baseAlertDialogBuilder.negativeText = getString(R.string.button_dismiss)
+        baseAlertDialogBuilder.positiveText = getString(R.string.button_share)
+        baseAlertDialogBuilder.positiveAction = {
+            createAndLaunchShareIntent(xpubWithCreationDate)
+            Unit
+        }
+        alertDialog = baseAlertDialogBuilder.buildAlertDialog()
+        alertDialog.show()
+
+        AdaptiveDialog
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/ToolsActivity.kt:129]: You lost this :) Shouldn't affect anything I guess

**Review by Syn-McJ — APPROVED** [review]
Looks good

File: wallet/src/de/schildbach/wallet/ui/ToolsActivity.kt (hunk 2 above)
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/ToolsActivity.kt:129]: Thanks for finding this, I was considering how I might use an `AdaptiveDialog` here, but forgot to delete this line.  I will delete it and get a build for QA.

**Review by Syn-McJ — APPROVED** [review]

**Pull request merged**